src/mmw/js/src/data_catalog/models.js

```diff
@@ -5,74 +5,102 @@ var _ = require('underscore'),
     App = require('../app'),
     utils = require('./utils');
 
-var DESCRIPTION_LENGTH = 300;
+var DESCRIPTION_MAX_LENGTH = 100;
 
-var DataCatalogModel = Backbone.Model.extend({
+var Catalog = Backbone.Model.extend({
     defaults: {
-        catalog: '',
-        isSearching: false,
-        searchText: ''
+        id: '',
+        name: '',
+        description: '',
+        loading: false,
+        active: false,
+        // Results collection
+        results: null,
+        resultCount: 0
+    },
+
+    search: function(query) {
+        var bounds = App.getLeafletMap().getBounds(),
+            bbox = utils.formatBounds(bounds),
+            data = {
+                catalog: this.id,
+                query: query,
+                bbox: bbox
+            };
+        return this.startSearch(data)
```

Comment:
One short-coming of this approach is that in-progress search requests are not cancelled.  I decided not to address this here.